File: 2022/PythonSolutions/Day10.py
def tick(cycle, crt, x_pos, r2):
    r2[crt // 40][crt % 40] = '#' if abs(x_pos - crt % 40) < 2 else '.'
    return (cycle % 40 == 20) * cycle * x_pos


with open("../Inputs/InputDay10.txt") as file:
    x, pos, c, res1 = 1, -1, 0, 0
    res2 = [[''] * 40 for _ in range(6)]
    for line in file:
        res1 += tick(c := c + 1, pos := pos + 1, x, res2)
        if line.startswith('add'):
            res1 += tick(c := c + 1, pos := pos + 1, x, res2)
            x += int(line.split()[1])
    print(res1, '\n'.join(''.join(x) for x in res2), sep='\n')

File: 2022/rust_solutions/src/day10.rs
use std::fs;

pub fn solve() {
    let data = fs::read_to_string("../Inputs/InputDay10.txt").expect("Unable to read file");
    let (mut x, mut pos, mut c, mut res1) = (1, 0, 0, 0);
    let mut res2 = [[' '; 40]; 6];
    for line in data.split('\n') {
        res1 += tick({c += 1; c}, &mut pos, x, &mut res2);
        if line.starts_with("add") {
            res1 += tick({c += 1; c}, &mut pos, x, &mut res2);
            x += line.split(' ').last().unwrap().parse::<i32>().unwrap();
        }
    }
    println!("{res1}\n{}", res2.map(|x| x.iter().collect::<String>()).join("\n"))
}

fn tick(cycle: i32, crt: &mut usize, x_pos: i32, r2: &mut [[char; 40]; 6]) -> i32 {
    r2[*crt / 40][*crt % 40] = if (x_pos - *crt as i32 % 40).abs() < 2 { '#' } else { '.' };
    *crt += 1;
    (cycle % 40 == 20) as i32 * cycle * x_pos
}